src/solvers/mod.rs

```diff
@@ -101,7 +101,7 @@ pub trait SolverModel {
     /// The type of the solution to the problem
     type Solution: Solution;
     /// The error that can occur while solving the problem
-    type Error: Debug;
+    type Error: 'static + Debug + std::error::Error;
```

Comment:
```suggestion
    type Error: std::error::Error;
```

`Debug` is already implied by `std::error::Error`